### PR TITLE
Bug 1714158: Prevent cert-syncer to act on stale data

### PR DESCRIPTION
**Pull request by tnozicka**
4.1 backport of https://github.com/openshift/library-go/pull/424

/cc @sttts

### CODE REVIEW

**Comment by tnozicka** [comment]
fake bump in https://github.com/openshift/cluster-kube-apiserver-operator/pull/557

File: pkg/operator/staticpod/certsyncpod/certsync_controller.go [hunk 1]
```diff
@@ -79,18 +85,69 @@ func (c *CertSyncController) sync() error {
 		case apierrors.IsNotFound(err) && !cm.Optional:
 			errors = append(errors, err)
 			continue
+
 		case apierrors.IsNotFound(err) && cm.Optional:
+			// Check with the live call it is really missing
+			configMap, err = c.configmapGetter.Get(cm.Name, metav1.GetOptions{})
+			if err == nil {
+				klog.Infof("Caches are stale. They don't see configmap '%s/%s', yet it is present", configMap.Namespace, configMap.Name)
```
Comment (sttts) [inline pkg/operator/staticpod/certsyncpod/certsync_controller.go]: isn't this something completely normal? Would at least use `V(2)`.

Comment (tnozicka) [reply pkg/operator/staticpod/certsyncpod/certsync_controller.go]: possibly, I don't want to touch it for backport

**Comment by mfojtik** [comment]
This met all formal criteria for a backport approval.

**Pull request merged**